### PR TITLE
policy: Do not make initial endpoint DROP_ALL mode dependent on polic…

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

File: pkg/endpoint/bpf.go [hunk 1]
```diff
@@ -218,12 +218,9 @@ func (e *Endpoint) writeHeaderfile(prefix string, owner Owner) error {
...
 	fw.WriteString(" */\n\n")
 
-	// If there hasn't been a policy calculated yet, we need to be sure we drop
-	// all packets, but only if policy enforcement
-	// is enabled for the endpoint / daemon.
-	if !e.PolicyCalculated &&
-		(e.Opts.IsEnabled(OptionIngressPolicy) || e.Opts.IsEnabled(OptionEgressPolicy)) &&
-		owner.PolicyEnforcement() != NeverEnforce {
+	// If policy has not been derived or calculated yet, all packets must
+	// be done until we have determined the policy of the endpoint.
```
Comment (ianvernon) [inline pkg/endpoint/bpf.go]: nit: done --> dropped

Comment (manalibhutiyani) [inline pkg/endpoint/bpf.go]: nit: `done` --> `dropped`

Comment (manalibhutiyani) [reply pkg/endpoint/bpf.go]: yikes :) Ian got there before me

Comment (manalibhutiyani) [reply pkg/endpoint/bpf.go]: Quick question: This enforces dropping on all endpoints, independent of whether a particular endpoint might actually end up not having a matching policy i.e we might drop traffic initially for a `endpoint` which does not have any matching `ingress` or `egress` policy after the policy is finally calculated? Wondering if that's the intended behavior here?

Comment (manalibhutiyani) [reply pkg/endpoint/bpf.go]: Nevermind, just saw your commit message. looks good.

**Review by manalibhutiyani — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
It is quite likely that this will cause CI to fail more likely if we don't wait for policy to be installed correctly

**Comment by eloycoto** [comment]
Backport: https://github.com/cilium/cilium/pull/3638

**Pull request merged**